src/shared/resourcefetcher/httpResourceFetcher.ts

```
@@ -3,45 +3,59 @@
      */
     public async get(): Promise<string | undefined> {
         try {
-            this.logger.verbose(`Loading resource from ${this.url}`)
+            this.logger.verbose(`Loading ${this.logText()}`)
 
-            const contents = await this.loadFromUrl()
+            const contents = (await this.getResponseFromGetRequest()).body
 
-            this.logger.verbose(`Finished loading resource from ${this.url}`)
+            this.logger.verbose(`Finished loading ${this.logText()}`)
 
             return contents
         } catch (err) {
-            this.logger.error(`Error loading resource from ${this.url}: %O`, err as Error)
+            this.logger.error(`Error loading ${this.logText()}: %O`, err as Error)
 
             return undefined
         }
     }
 
-    private async loadFromUrl(): Promise<string | undefined> {
-        return new Promise<string>((resolve, reject) => {
-            request(this.url, (error: any, response: IncomingMessage, body: any) => {
-                if (error) {
-                    reject(error)
-                } else {
-                    // tslint:disable-next-line: no-unsafe-any
-                    resolve(body.toString())
+    // TODO: Are there cases where we don't mind the URL?
+    // Safer to do it this way assuming others use this fetcher.
+    private logText(): string {
+        return this.params.showUrl ? this.url : this.params.friendlyName ?? 'resource from URL'
+    }
+
+    private async getResponseFromGetRequest(): Promise<request.Response> {
+        return new Promise<request.Response>((resolve, reject) => {
+            const call = request(this.url, (err, response, body) => {
+                if (err) {
+                    // swallow error to keep URL private
+                    // some AWS APIs use presigned links (e.g. Lambda.getFunction); showing these represent a securty concern.
+                    reject('Error making request to URL.')
```

err is eaten, consider outputting err.code (if available) or relevant but safe status fields in the reject call

---

Code might work. Message and stack both contain the URL.